pkg/glanceapi/statefulset.go

```diff
@@ -290,19 +296,32 @@ func StatefulSet(
 		extraVolPropagation),
 		apiVolumes...)
 
-	// If possible two pods of the same service should not
-	// run on the same worker node. If this is not possible
-	// the get still created on the same worker node.
-	statefulset.Spec.Template.Spec.Affinity = affinity.DistributePods(
-		common.AppSelector,
-		[]string{
-			glance.ServiceName,
-		},
-		corev1.LabelHostname,
-	)
 	if instance.Spec.NodeSelector != nil {
 		statefulset.Spec.Template.Spec.NodeSelector = *instance.Spec.NodeSelector
 	}
 
+	if topologyOverride != nil {
+		ts := topologyOverride.Spec
+		if ts.TopologySpreadConstraint == nil {
+			ts.TopologySpreadConstraint = &tplg
+		} else {
+			statefulset.Spec.Template.Spec.TopologySpreadConstraints = *topologyOverride.Spec.TopologySpreadConstraint
+		}
+		if ts.Affinity == nil {
+			// If possible two pods of the same service should not
+			// run on the same worker node. If this is not possible
+			// the get still created on the same worker node.
+			statefulset.Spec.Template.Spec.Affinity = affinity.DistributePods(
+				common.AppSelector,
+				[]string{
+					glance.ServiceName,
+				},
+				corev1.LabelHostname,
+			)
+		} else {
+			statefulset.Spec.Template.Spec.Affinity = ts.Affinity
+		}
+	}
+
 	return statefulset, err
```

Comment:
do we need to set topology on the other PodSpecTemplates too? e.g Jobs

Reply:
hrmm, thinking about it further we shouldn't, at least for now....  The job pods definitely should not be considered in the same topologyconstraint rule as the statefulset or they would mess up the count for skew etc...